makelib/golang.mk

```diff
@@ -84,10 +84,14 @@ GOIMPORTS := $(TOOLS_HOST_DIR)/goimports
 GO := go
 GOHOST := GOOS=$(GOHOSTOS) GOARCH=$(GOHOSTARCH) go
 GO_VERSION := $(shell $(GO) version | sed -ne 's/[^0-9]*\(\([0-9]\.\)\{0,4\}[0-9][^.]\).*/\1/p')
+GO_REQUIRED_VERSION := 1.18
+ifneq ($(GO_VERSION),$(GO_REQUIRED_VERSION))
```

Comment:
We used to have something like this but the friction it creates turned out to be more than the benefits we're getting for streamlining. So, I'd opt to let individual projects impose this requirement if they'd like to in their `Makefile`s.

Reply:
Made `GO_REQUIRED_VERSION` overridable in https://github.com/upbound/build/pull/195/commits/09af1abbfb10263202cd9a03893d63054d53b8e0